### PR TITLE
Api docs for HttpOverrides

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Part of #26397.

Breaking it up into smaller PRs for review. This one covers HttpOverrides.

### CODE REVIEW

File: src/Middleware/HttpOverrides/src/ForwardedHeaders.cs [hunk 1]
```diff
@@ -5,13 +5,31 @@
...
 namespace Microsoft.AspNetCore.HttpOverrides
 {
+    /// <summary>
+    /// Flags for controlling which forwarders are processed.
+    /// </summary>
     [Flags]
     public enum ForwardedHeaders
     {
+        /// <summary>
+        /// Do not processed any forwarders
```
Comment (sharkk121) [inline src/Middleware/HttpOverrides/src/ForwardedHeaders.cs:15]:
```suggestion
        /// Do not process any forwarders
```

File: src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs [hunk 2]
```diff
@@ -7,15 +7,22 @@
...
+    /// <summary>
+    /// Extension methods for enabling <see cref="ForwardedHeadersMiddleware"/>.
+    /// </summary>
     public static class ForwardedHeadersExtensions
     {
         private const string ForwardedHeadersAdded = "ForwardedHeadersAdded";
 
         /// <summary>
-        /// Forwards proxied headers onto current request
+        /// Forwards proxied headers onto current request.
```
Comment (pranavkm) [inline src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs:18]: I tweaked this file a bit. Can you have a look?

Comment (Tratcher) [reply src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs:18]:
```suggestion
        /// Applies forwarded headers to their matching fields on the current request.
```

File: src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs [hunk 3]
```diff
@@ -7,15 +7,22 @@
...
+        /// <para>
+        /// By convention, HTTP proxies forward information from the client in well-known HTTP headers.
+        /// The Forwarded Headers Middleware, reads these headers and fills in the associated fields on HttpContext.
```
Comment (Tratcher) [inline src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs:21]:
```suggestion
        /// The <see cref="ForwardedHeadersMiddleware"> reads these headers and fills in the associated fields on HttpContext.
```

File: src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs [hunk 4]
```diff
@@ -7,24 +7,27 @@
...
 namespace Microsoft.AspNetCore.Builder
 {
+    /// <summary>
+    /// Options for <see cref="ForwardedHeadersMiddleware"/>
+    /// </summary>
     public class ForwardedHeadersOptions
     {
         /// <summary>
-        /// Use this header instead of <see cref="ForwardedHeadersDefaults.XForwardedForHeaderName"/>
+        /// Gets or sets the header used to retrieve the originating client IP. Defaults to the value specified by
```
Comment (pranavkm) [inline src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs:16]: Also edited this.

File: src/Middleware/HttpOverrides/src/HttpMethodOverrideMiddleware.cs [hunk 5]
```diff
@@ -29,6 +37,11 @@ public HttpMethodOverrideMiddleware(RequestDelegate next, IOptions<HttpMethodOve
             _options = options.Value;
         }
 
+        /// <summary>
+        /// Executes the middleware.
+        /// </summary>
+        /// <param name="context">The <see cref="HttpContext"/> for the current request.</param>
+        /// <returns>A task that represents the execution of this middleware.</returns>
```
Comment (pranavkm) [inline src/Middleware/HttpOverrides/src/HttpMethodOverrideMiddleware.cs:44]:
```suggestion
```

File: src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs [hunk 6]
```diff
@@ -36,10 +43,14 @@ public static IApplicationBuilder UseForwardedHeaders(this IApplicationBuilder b
 
         /// <summary>
         /// Forwards proxied headers onto current request
+        /// <para>
+        /// By convention, HTTP proxies forward information from the client in well-known HTTP headers.
+        /// The Forwarded Headers Middleware, reads these headers and fills in the associated fields on HttpContext.
```
Comment (Tratcher) [inline src/Middleware/HttpOverrides/src/ForwardedHeadersExtensions.cs:48]:
```suggestion
        /// The <see cref="ForwardedHeadersMiddleware"> reads these headers and fills in the associated fields on HttpContext.
```

File: src/Middleware/HttpOverrides/src/HttpMethodOverrideExtensions.cs [hunk 7]
```diff
@@ -7,6 +7,9 @@
 
 namespace Microsoft.AspNetCore.Builder
 {
+    /// <summary>
+    /// Extension methods for enabling <see cref="HttpMethodOverrideMiddleware"/>.
+    /// </summary>
     public static class HttpMethodOverrideExtensions
```
Comment (pranavkm) [inline src/Middleware/HttpOverrides/src/HttpMethodOverrideExtensions.cs:13]: Can we add a snippet to the description of the `Use*` methods that say when you would use this feature? Maybe something like 

> This middleware is used when a client wants to express arbitrary HTTP methods but is limited to sending GET or POST methods. By default, the `X-HTTP-Method-Override` request header is used to specify the HTTP method being tunneled.

File: src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs [hunk 8]
```diff
@@ -7,24 +7,27 @@
...
+        /// Gets or sets the header used to retrieve the original value of the Host header field.
+        /// Defaults to the value specified by <see cref="ForwardedHeadersDefaults.XForwardedHostHeaderName"/>
         /// </summary>
-        /// <seealso cref="ForwardedHeadersDefaults"/>
         public string ForwardedHostHeaderName { get; set; } = ForwardedHeadersDefaults.XForwardedHostHeaderName;
 
         /// <summary>
-        /// Use this header instead of <see cref="ForwardedHeadersDefaults.XForwardedProtoHeaderName"/>
+        /// Gets or sets the header used to retrieve the value for the originating scheme (HTTP/HTTPS).
+        /// Defaults to the value specified by <see cref="ForwardedHeadersDefaults.XForwardedProtoHeaderName"/>
```
Comment (Tratcher) [inline src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs:29]: Why not make the same changes to the rest of this file?

**Review by pranavkm — COMMENTED** [review]
I tweaked the

File: src/Middleware/HttpOverrides/src/IPNetwork.cs [hunk 9]
```diff
@@ -26,6 +37,11 @@ public IPNetwork(IPAddress prefix, int prefixLength)
 
         private byte[] Mask { get; }
 
+        /// <summary>
+        /// Determine whether a given The <see cref="IPAddress"/> is part of the IP network.
+        /// </summary>
+        /// <param name="address">The <see cref="IPAddress"/>.</param>
+        /// <returns><c>true</c> if the <see cref="IPAddress"/> is part of the IP network. Otherwise, <c>false</c>.</returns>
```
Comment (pranavkm) [inline src/Middleware/HttpOverrides/src/IPNetwork.cs:44]: I think we use `<see langword="true" />` in more places, so that wins.

**Comment by JunTaoLuo** [comment]
:up::date:

File: src/Middleware/HttpOverrides/src/HttpMethodOverrideExtensions.cs (hunk 7 above)
Comment (JunTaoLuo) [reply src/Middleware/HttpOverrides/src/HttpMethodOverrideExtensions.cs:13]: Updated

File: src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs (hunk 8 above)
Comment (JunTaoLuo) [reply src/Middleware/HttpOverrides/src/ForwardedHeadersOptions.cs:29]: Updated the other header names. Did you want more details for everything else too? I feel the others didn't quite need as much elaboration.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**